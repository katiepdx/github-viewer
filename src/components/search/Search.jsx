import React from 'react'
import PropTypes from 'prop-types'
import styles from '../app/App.css'

const Search = ({ onChange, onClick }) => {
  return (
    <>
      <div className={styles.searchBar}>
        <label htmlFor="ghUser">Search</label>
        <input
          id="ghUser"
          type="text"
          name="ghUser"
          onChange={onChange}
        />

        <button onClick={onClick}>Go!</button>
      </div>
    </>
  )
}

Search.propTypes = {
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired
}

export default Search
