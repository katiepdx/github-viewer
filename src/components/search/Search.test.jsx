import React from 'react'
import { render } from '@testing-library/react'
import Search from './Search'

describe('<Search/>', () => {
  it('renders the search bar and go button', () => {
    const { asFragment } = render(<Search
      onChange={() => { }}
      onClick={() => { }}
    />)

    expect(asFragment()).toMatchSnapshot()
  })
})
