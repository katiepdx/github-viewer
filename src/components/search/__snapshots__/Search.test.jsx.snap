// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> renders the search bar and go button 1`] = `
<DocumentFragment>
  <div
    class="searchBar"
  >
    <label
      for="ghUser"
    >
      Search
    </label>
    <input
      id="ghUser"
      name="ghUser"
      type="text"
    />
    <button>
      Go!
    </button>
  </div>
</DocumentFragment>
`;
